### PR TITLE
FitRideFile.cpp function RideFile * run() can access deleted swimXdata pointer

### LINKED ISSUE #3740: FitRideFile.cpp function  RideFile * run() can access deleted swimXdata pointer
In FitRideFile.cpp
    function  RideFile * run()

It would appear that the swimXdata test after appendXdata call is unecessary, and to make matters worse swimXdata can be deleted in appendXdata leading to random crashes on imports.

appendXdata(rideFile)

if (!swimXdata->datapoints.empty()) {
                // Build synthetic kph, km and cad sample data for Lap Swims
                DataProcessor* fixLapDP = DataProcessorFactory::instance().getProcessors(true).value("Fix Lap Swim");
                if (fixLapDP) fixLapDP->postProcess(rideFile, NULL, "NEW");
                else qDebug()<<"Fix Lap Swim Data Processor not found.";
            }

I have summitted a fix to test the swimXdata pointer, but now I'm wondering why the test is required in the first place.

### PR DESCRIPTION
**Pull request by scr142**
resolves #3740

In FitRideFile.cpp
function RideFile * run()

It would appear that the swimXdata test after appendXdata call is unecessary, and to make matters worse swimXdata can be deleted in appendXdata leading to random crashes on imports.

appendXdata(rideFile)

if (!swimXdata->datapoints.empty()) {
// Build synthetic kph, km and cad sample data for Lap Swims
DataProcessor* fixLapDP = DataProcessorFactory::instance().getProcessors(true).value("Fix Lap Swim");
if (fixLapDP) fixLapDP->postProcess(rideFile, NULL, "NEW");
else qDebug()<<"Fix Lap Swim Data Processor not found.";
}

I have summitted a fix to test the swimXdata pointer, but now I'm wondering why the test is required in the first place.

### CODE REVIEW

File: src/FileIO/FitRideFile.cpp [hunk 1]
```diff
@@ -3280,16 +3279,19 @@ struct FitFileReaderState
...
             delete swimXdata;
+            swimXdata = nullptr;
 
         if (!hrvXdata->datapoints.empty())
             rf->addXData("HRV", hrvXdata);
         else
             delete hrvXdata;
+            hrvXdata = nullptr;
 
         if (!gearsXdata->datapoints.empty())
```
Comment (scr142) [inline src/FileIO/FitRideFile.cpp:3296]: Should also set the gearsXdata pointer to nullptr after its deleted

Comment (scr142) [reply src/FileIO/FitRideFile.cpp:3296]: I'm sorry about the three commits, I realised I'd left a comment to my self in the code and wanted to remove it.  I'm a newbie to Git (I've used SCCS & SVN in the distant past) and couldn't see how to delete a commit to my repository.  If there is a more elegant method to fix this issue then please discard this request, although it has now made my importation of fit files stable now.

Comment (amtriathlon) [reply src/FileIO/FitRideFile.cpp:3296]: Thanks for the catch, I will add a simpler fix checking ridefile has SWIM XData series before to call Fix Lap Swim DP, which was the intention of the original (erroneous) test.

Comment (scr142) [reply src/FileIO/FitRideFile.cpp:3296]: Thank you

**Comment by amtriathlon** [comment]
Replaced by c6c5505610ba5de93cd83e5992ff321028343331

**Pull request closed without merging**